clippy_lints/src/pub_underscore_fields.rs

```diff
@@ -0,0 +1,57 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+use rustc_ast::ast::{Item, ItemKind, VisibilityKind};
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks whether any field of the struct is prefixed with an `_` (underscore) and also marked
+    /// `pub` (public)
+    ///
+    /// ### Why is this bad?
+    /// Fields prefixed with an `_` are inferred as unused, which suggests it should not be marked
+    /// as `pub`, because marking it as `pub` infers it will be used.
+    ///
+    /// ### Example
+    /// ```rust
+    /// struct FileHandle {
+    ///     pub _descriptor: usize,
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// struct FileHandle {
+    ///     _descriptor: usize,
+    /// }
+    /// ```
```

Comment:
The other alternative is also needed :cat2:

```suggestion
    /// ```rust
    /// struct FileHandle {
    ///     _descriptor: usize,
    /// }
    /// // OR 
    /// struct FileHandle {
    ///     pub descriptor: usize,
    /// }
    /// ```
```

Reply:
Ah ha, so right, updated!